plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -643,6 +651,36 @@ private boolean deleteObject(String key)
         }
     }
 
+    @Override
+    public void deleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            Iterator<List<Path>> partitions = Iterables.partition(paths, BATCH_DELETE_PARTITION_SIZE).iterator();
+            while (partitions.hasNext()) {
+                List<Path> currentBatch = partitions.next();
+                deletePaths(currentBatch);
+            }
+        }
+        catch (AmazonClientException e) {
+            throw new IOException("Exception while batch deleting paths from s3", e);
```

Comment:
remove "from s3" (since it's kind of obvious; or you'd capitalize S3 since it's a product name)